web-registry/src/components/organisms/CreditClassForms/CreateCreditClassForm.tsx

```diff
@@ -0,0 +1,93 @@
+import { Formik } from 'formik';
+
+import SaveFooter from 'web-components/lib/components/fixed-footer/SaveFooter';
+import NotFound from 'web-components/lib/components/not-found';
+import RotationalGrazing from 'assets/rotational-grazing.png';
+
+import { useMultiStep } from 'components/templates/MultiStepTemplate';
+import { CreditClassForm } from './CreditClassForm';
+import { CreditClassReview } from './CreditClassReview';
+import { CreditClassFinished } from './CreditClassFinished';
+import { mockTxHash, mockClassId } from './CreditClass.mocks';
+
+import type { CreditClassValues } from './CreditClassForm';
+
+export const createCreditClassSteps = [
+  {
+    id: 'create-credit-class',
+    name: 'Create Credit Class',
+    title: 'Create Credit Class',
+  },
+  {
+    // TODO: ID probably isn't needed on the next two, as they aren't stored in
+    // localstorage?
+    id: '',
+    name: 'Review',
+    title: 'Review',
+  },
+  {
+    id: '',
+    name: 'Finished',
+    title: 'Finished',
+  },
+];
+
+/** must be used within a `MultiStep` context  */
+export const CreateCreditClassForm = (props: {
+  onSubmit: (values: CreditClassValues) => void;
+}): JSX.Element => {
+  const {
+    data,
+    percentComplete,
+    activeStep,
+    handleBack,
+    handleNext,
+    isLastStep,
+  } = useMultiStep<CreditClassValues>();
+
+  const disabledFields: string[] = [];
+  if (!!data.admin.length) disabledFields.push('admin');
+  if (!!data.fee.length) disabledFields.push('fee');
+
+  const CurrentStep = (): React.ReactElement => {
```

Comment:
I tend to agree with @flagrede here - this was a bit of an oversight for something I do quickly during dev out of habit, then later overlook. It was generally okay to do in react class components from what I understand, because only the render function would change. 

Function components are different though. With this structure, `CurrentStep` will be defined each time each time `CreateCreditClassForm` re-renders -- which shouldn't be often, but still a simple problem to avoid. 

I've moved this to a simple non-exported component which lives in the same file 👍